src/sentry/api/endpoints/organization_events_meta.py

```diff
@@ -88,6 +88,94 @@ def get(self, request, organization):
         return Response(baseline_data)
 
 
+class OrganizationEventsTrace(OrganizationEventsEndpointBase):
+    def get(self, request, organization, trace_id):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        # Get 100 events from snuba, and pray these cover the trace
+        with self.handle_query_errors():
+            result = discover.query(
+                selected_columns=[
+                    "id",
+                    "timestamp",
+                    "transaction",
+                    "project_id",
+                    "trace.span",
+                    "trace.parent_span",
+                ],
+                orderby=["-timestamp", "trace.parent_span", "id"],
+                params=params,
+                query=f"event.type:transaction trace:{trace_id}",
+                limit=100,
+                referrer="experimental.api.trace-view.get_ids",
+            )
+            if len(result["data"]) == 0:
+                return Response(status=404)
+
+        # Get all our events
+        events = {
+            event.event_id: event
+            for event in eventstore.get_events(
+                eventstore.Filter(
+                    project_ids=params["project_id"],
+                    event_ids=[event["id"] for event in result["data"]],
+                )
```

Comment:
Would adding start and end timestamps as conditions help with fetching transactions faster? This could be derived from the snuba result set.

Reply:
TBH, less familiar with nodestore. But I imagine id should already be pretty fast. I'll experiment with this if it turns out to be an issue though :+1:

Reply:
Yep, just tested it and is pretty slow. I'll try adding a time period :+1:

Reply:
Nodestore doesn't use timestamps at all. It is a key-value store where the keys are [a checksum of the project and eventid](https://github.com/getsentry/sentry/blob/f2c2b833a8626716279cd4c3547e46cdce3434a9/src/sentry/eventstore/models.py#L258-L266). A date range will help the snuba search that is used to find records so that data can be loaded from nodestore.